lib/index.js

```diff
@@ -66,27 +66,32 @@ const internals = {
 };
 
 
-exports.Boom = class extends Error {
+exports.Boom = class Boom extends Error {
 
-    constructor(messageOrError, options = {}) {
+    data = null;
+    output;
 
-        if (messageOrError instanceof Error) {
-            return exports.boomify(Hoek.clone(messageOrError), options);
-        }
+    get isServer() {
+
+        return this.output.statusCode >= 500;
+    }
 
-        const { statusCode = 500, data = null, ctor = exports.Boom } = options;
-        const error = new Error(messageOrError ? messageOrError : undefined);         // Avoids settings null message
-        Error.captureStackTrace(error, ctor);                           // Filter the stack to our external API
-        error.data = data;
-        const boom = internals.initialize(error, statusCode);
+    set isServer(value) {}                                             // Allow for compatiblity with legacy boom
```

Comment:
Implementation might rely on being able to actually change this. It might be better to actually let an error propagate here:
```suggestion
    set isServer(value) {
      throw new TypeError('isServer is readonly')
    }
```

Reply:
Sure they might, but it is unlikely. They can still customise it through subclassing.

We cannot throw, since a new boom error passed to an old boomify will do a `boom.isServer = true/false` on our object.

I guess we could store the value when set, overriding the current descriptor:

```js
set isServer(value) {

    Object.defineProperty(this, 'isServer', {
        value,
        writable: true,
        configurable: true
    });
}
```

Reply:
I have implemented this in 1e0c4643a43c6d78cad169bd3b8e00010a903e12.